custom/icds_reports/reports/governance_apis.py

```diff
@@ -121,3 +121,47 @@ def base_data(row_data):
 def get_state_names():
     return list(AwcLocation.objects.filter(aggregation_level=AggregationLevels.STATE, state_is_test=0
                                            ).values('state_site_code', 'state_name'))
+
+
+@icds_quickcache(['length', 'year', 'month', 'order', 'query_filters'], timeout=30 * 60)
+def get_cbe_data(length, year, month, order, query_filters):
+    data = AggGovernanceDashboard.objects.filter(
+        month=date(year, month, 1),
+        **query_filters
+    ).order_by(*order).annotate(
+        cbe_conducted_1=Case(
+            When(cbe_date_1__isnull=False, then=Value('yes')),
+            default=Value('no'),
+            output_field=TextField(),
+        ),
+        cbe_conducted_2=Case(
+            When(cbe_date_2__isnull=False, then=Value('yes')),
+            default=Value('no'),
+            output_field=TextField(),
+        ),
+    ).values(
+        'awc_id',
+        'awc_code',
+        'cbe_conducted_1',
+        'cbe_type_1',
+        'cbe_date_1',
+        'num_target_beneficiaries_1',
+        'num_other_beneficiaries_1',
+        'cbe_conducted_2',
+        'cbe_type_2',
+        'cbe_date_2',
+        'num_target_beneficiaries_2',
+        'num_other_beneficiaries_2',
+    )
+
+    def get_value_or_data_not_entered(value):
+        if value is None:
+            return DATA_NOT_ENTERED
```

Comment:
Why send this string instead of null? It is much harder to parse this than a null value, especially since it means the type of the column can be different (ie it can be a date or a string rather than just a nullable date). Furthermore, this data is likely going into a data base so human readable strings do not really provide value. It also means a change to that constant, if we decide to change how the dashboard displays these will break everyones custom parsing.

Reply:
this same comment applies to the rest of these APIs as well

Reply:
I see. yeah, null makes more sense